features/issues/api/use-get-issues.tsx

```diff
@@ -2,32 +2,36 @@ import { useEffect } from "react";
 import { useQuery, useQueryClient } from "@tanstack/react-query";
 import { getIssues } from "@api/issues";
 import type { Page } from "@typings/page.types";
-import type { Issue } from "@api/issues.types";
+import type { Issue, IssuesParams } from "@api/issues.types";
 
 const QUERY_KEY = "issues";
 
-export function getQueryKey(page?: number) {
-  if (page === undefined) {
-    return [QUERY_KEY];
-  }
-  return [QUERY_KEY, page];
+export function getQueryKey(params: IssuesParams) {
+  const queryKey: (string | number)[] = [QUERY_KEY];
+  if (!params) return queryKey;
+  Object.values(params).map((param) => {
+    if (param) queryKey.push(param);
+  });
+  return queryKey;
 }
 
-export function useGetIssues(page: number) {
+export function useGetIssues(params: IssuesParams) {
   const query = useQuery<Page<Issue>, Error>(
-    getQueryKey(page),
-    ({ signal }) => getIssues(page, { signal }),
+    getQueryKey(params),
+    ({ signal }) => getIssues(params, { signal }),
     { keepPreviousData: true },
   );
 
   // Prefetch the next page!
   const queryClient = useQueryClient();
   useEffect(() => {
     if (query.data?.meta.hasNextPage) {
-      queryClient.prefetchQuery(getQueryKey(page + 1), ({ signal }) =>
-        getIssues(page + 1, { signal }),
+      const nextPage = params.page || 1 + 1;
+      queryClient.prefetchQuery(
+        getQueryKey({ ...params, page: nextPage }),
+        ({ signal }) => getIssues({ ...params, page: nextPage }, { signal }),
       );
     }
-  }, [query.data, page, queryClient]);
+  }, [query.data, params, queryClient]); // Passing in an object could cause errors here
```

Comment:
Yeah I think you might trigger the effect on each render because there's an object in the dependencies (at least if the object is recreated every time). You might use `...Object.values(params)` instead but again not sure if that returns an array with the same order every time